pandas/core/indexes/base.py

```diff
@@ -151,6 +151,11 @@ class Index(IndexOpsMixin, StringAccessorMixin, PandasObject):
 
     _engine_type = libindex.ObjectEngine
 
+    _accessors = frozenset(['dt', 'str', 'cat'])
```

Comment:
> I suppose we already have tests to check that 'str' is only present on the correct index types?

I'm not sure off the top, but I wrote some tests for #17204 that can be adapted pretty readily.